bin/selection_utils.py

```diff
@@ -161,58 +161,98 @@ def target_distance_fcn(args, this_seq, target_seqs):
     return itarget, tdist
 
 # ----------------------------------------------------------------------------------------
+def count_aa(aa_seq,which_aa):
+    list_of_aa = list(aa_seq)
+    res1 = []
+    for i in range(len(list_of_aa)):
+        if list_of_aa[i] == which_aa:
+            res1.append(i)
+    return len(res1)
+
+def count_not_that_aa(aa_seq,which_aa):
+    list_of_aa = list(aa_seq)
+    res1 = []
+    for i in range(len(list_of_aa)):
+        if list_of_aa[i] != which_aa:
+            res1.append(i)
+    return len(res1)
+
 def calc_kd(node, args):
     if has_stop_aa(node.aa_seq):  # nonsense sequences have zero affinity/infinite kd
         return float('inf')
     if not args.selection:
         return 1.
 
-    assert args.mature_kd < args.naive_kd
-    tdist = node.target_distance if args.min_target_distance is None else max(node.target_distance, args.min_target_distance)
-    kd = args.mature_kd + (args.naive_kd - args.mature_kd) * (tdist / float(args.target_distance))**args.k_exp  # transformation from distance to kd
-
+    #tdist = node.target_distance if args.min_target_distance is None else max(node.target_distance, args.min_target_distance)
+    #kd = args.mature_kd + (args.naive_kd - args.mature_kd) * (tdist / float(args.target_distance))**args.k_exp  # transformation from distance to kd
+    #kd = 100 - 3*min(count_not_that_aa(node.aa_seq,'A'),33)
+    kd = 100 - 12*min(count_aa(node.aa_seq,'L'),9)
```

Comment:
This seems like it completely changes the way kd is calculated? My understanding was that we were going to add some new options that allowed to turn on new features, but not modify existing behavior.

Reply:
As you can see by all the commented out portions, this was my rough initial testing of some things. Perhaps the best way to implement it would be flag (i.e. --torchDMS) so that if --torchDMS "true" was passed, it would used a torchDMS prediction, if --neutral were passed it would return some nM affinity for every value that allows for GC expansion (i.e. lower than 100 nM), and if --target_distance were passed it returns the target distance based Kd. Does that sound good?

Reply:
ah ok, sorry, I misunderstood -- we should get this finished up before merging the pull request. Yes, something like that sounds great. I think the default should remain unchanged (i.e if someone runs with the same command line args before and after we make these changes, it's important they get identical results). I think a string flag with several choices ([like this](https://github.com/matsengrp/bcr-phylo-benchmark/blob/master/bin/simulator.py#L804)) might be the way to go, perhaps with a name like --affinity-to-fitness-mapping?